source/common/formatter/stream_info_formatter.cc

```diff
@@ -1085,6 +1086,28 @@ const StreamInfoFormatterProviderLookupTable& getKnownStreamInfoFormatterProvide
                     return absl::make_optional<std::string>(std::move(host_name));
                   });
             }}},
+          {"UPSTREAM_HOST_NAME_WITHOUT_PORT",
+           {CommandSyntaxChecker::COMMAND_ONLY,
+            [](absl::string_view, absl::optional<size_t>) {
+              return std::make_unique<StreamInfoStringFormatterProvider>(
+                  [](const StreamInfo::StreamInfo& stream_info) -> absl::optional<std::string> {
+                    const auto opt_ref = stream_info.upstreamInfo();
+                    if (!opt_ref.has_value()) {
+                      return absl::nullopt;
+                    }
+                    const auto host = opt_ref->upstreamHost();
+                    if (host == nullptr) {
+                      return absl::nullopt;
+                    }
+                    std::string host_name = host->hostname();
+                    if (host_name.empty()) {
+                      // If no hostname is available, the main address is used.
+                      return host->address()->ip()->addressAsString();
```

Comment:
I think this will crash if the upstream address is not an IP. It's not common, but this could be a named pipe or an envoy internal address. The various implementations are here: https://github.com/envoyproxy/envoy/blob/main/source/common/network/address_impl.h

I recommend just setting `host_name` to `host->address()->asString()` and then let the port stripping happen below.